src/TextArea.js

```diff
@@ -0,0 +1,246 @@
+        }
+    }
+
+    shouldDoAutoGrow() {
+        return this.props.autoGrow && !this.userHasResized
+    }
+
+    hasUserResized() {
+        const { width: oldWidth, height: oldHeight } = this.textareaDimensions
+
+        this.setTextareaDimensions()
+
+        const { width: newWidth, height: newHeight } = this.textareaDimensions
+        const userHasResized = newWidth !== oldWidth || newHeight !== oldHeight
+
+        if (userHasResized) {
+            this.userHasResized = true
+            this.removeResizeListener()
+            this.setHasScrollBar()
+        }
+    }
+
+    getIconRightPosition() {
+        const base = 2
+        const right = this.state.hasScrollbar ? base + this.scrollbarSize : base
+        return `${right}px`
+    }
+
+    render() {
+        const {
+            className,
+            onChange,
+            onFocus,
+            onBlur,
+            dense,
+            disabled,
+            readOnly,
+            placeholder,
+            name,
+            valid,
+            error,
+            warning,
+            loading,
+            value,
+            tabIndex,
+            rows,
+            width,
+            resize,
+        } = this.props
+
+        return (
+            <div className={cx('container', className)}>
+                <textarea
+                    id={name}
+                    name={name}
+                    placeholder={placeholder}
+                    ref={this.textareaRef}
+                    value={value}
+                    disabled={disabled || readOnly}
+                    tabIndex={tabIndex}
+                    onFocus={onFocus}
+                    onBlur={onBlur}
+                    onChange={onChange}
+                    rows={rows}
+                    className={cx('textarea', className, {
+                        dense,
+                        disabled,
+                        error,
+                        valid,
+                        warning,
+                        'read-only': readOnly,
+                    })}
+                />
+
+                <div className="status-icon">
+                    <StatusIconNoDefault
+                        error={error}
+                        valid={valid}
+                        loading={loading}
+                        warning={warning}
+                    />
+                </div>
+
+                <style jsx>{styles}</style>
+                <style jsx>{`
+                    .textarea {
+                        width: ${width};
+                        height: ${this.state.height};
+                        resize: ${resize};
+                    }
+                    .status-icon :global(svg) {
+                        right: ${this.getIconRightPosition()};
+                    }
+                `}</style>
+            </div>
+        )
+    }
+}
+
+TextArea.defaultProps = {
+    rows: 4,
+    width: '100%',
+    resize: 'vertical',
+    onBlur: () => {},
+    onFocus: () => {},
```

Comment:
Is this necessary? I think it should be ok to just leave this out, as it's just passed down to the `textarea`, which has its own default prop for `onBlur` and `onFocus`